File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int n = x;
        int reversedNum = 0;
        while(n>0)
        {
            int lastDigit = n%10;
            reversedNum = reversedNum * 10 + lastDigit;
            n=n/10;
        }
        if(reversedNum == x) return true;
        return false;     
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0) return false; 
        long long reversedNum = 0;
        int originalNum = x;
        while(x!=0)
        {
            reversedNum = reversedNum * 10 + x % 10;
            x = x / 10;
        }
        return reversedNum == originalNum;
    }
};